app/models/order_item.rb

```diff
@@ -1,37 +1,70 @@
 class OrderItem < ActiveRecord::Base
-  # DB relationships
-  belongs_to :product
   belongs_to :order
+  belongs_to :product
+  has_one :seller, through: :product
+
+  before_save :product_has_stock?
 
-  # data validations
   validates :product_id, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :order_id, presence: true, numericality: { only_integer: true, greater_than: 0 }
   validates :quantity_ordered, presence: true, numericality: { only_integer: true, greater_than: 0 }
-  # !R !W is there a way we can adjust this validation to check the product first?
+  validates :status, presence: true, inclusion: { in: %w(pending paid shipped canceled),
+    message: "%{value} is not a valid status" }
+  validate :order_item_is_unique?, on: [:create]
+
+  def more!
+    if product_has_stock? && product.stock > quantity_ordered
+      increment!(:quantity_ordered, 1)
+    else
+      errors.add(:product_stock, "Product must have available stock.")
+    end
+  end
+
+  def less!
+    if quantity_ordered > 1
+      update_column(:quantity_ordered, quantity_ordered - 1)
+    else
+      errors.add(:quantity_ordered, "You must remove this from your cart if you want to reduce its quantity any further.")
```

Comment:
There are a number of error message strings used in your models, these would be great to put into constants and/or all in one array to keep separate from the rest of your code.